### DISCUSSION TITLE
Auto Paginate

### DISCUSSION THREAD

**Discussion by robokozo** [vuejs/vuefire#1243] [Ideas]
I haven't had a chance to try out @next yet but I'm exploring the new docs and it seems the 'way' to hook up a query is to `useList`. Is that right?

One thing that I'd really like to automate and simplify is the process of paginating data... something like

```ts
const myQuery = query(productsRef,  orderBy("dateAdded"),  startAfter(lastSnapshot), limit(10));
```

it would be great if I could do something like
```ts
const myQuery = query(productsRef, orderBy("dateAdded"));

const { data, nextPage } = useList(query, {
  offset: 100,
  limit: 10
})

watch(scrolledAllTheWayDown, () => nextPage())
```

Sorry for the half baked idea, I just thought since pagination is quite a common strategy for keeping firestore cost to a minimum, it'd be really nice to offer a nice api to simplify.

**Comment by posva** [comment]
yes, `useList()` for Database and `useCollection()` for Firestore

I think that once vue refs can be passed to `useList()` and others, creating a computed of the query would just work ™️ .

But ideally, we could expose something extra that exposes controls like nextPage and previousPage (since we can hook into the data snapshots). It would require having a way of passing reactive limits and offsets too, so it's still not sure which one of the versions would be easier to use
